module/VuFind/src/VuFind/Recommend/Databases.php

```diff
@@ -0,0 +1,275 @@
+use function intval;
+
+/**
+ * Databases Recommendations Module
+ *
+ * This class displays a list of external links to the research databases represented
+ * by EDS or similar results.  (Unlike the EDS ContentProvider facet that would narrow
+ * down the results within VuFind.)
+ *
+ * @category VuFind
+ * @package  Recommendations
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+class Databases implements RecommendInterface
+{
+    use \VuFind\Cache\CacheTrait;
+
+    /**
+     * Results object
+     *
+     * @var \VuFind\Search\Base\Results
+     */
+    protected $results;
+
+    /**
+     * Configuration manager
+     *
+     * @var ConfigManager
+     */
+    protected $configManager;
+
+    /**
+     * Number of results to show
+     *
+     * @var int
+     */
+    protected $limit;
+
+    /**
+     * Name of the configuration file for databases config (minus ".ini").
+     *
+     * @var string
+     */
+    protected $databasesConfigFile;
+
+    /**
+     * The result facet with the list of databases.  Each value in the
+     * array is a level of the facet hierarchy.
+     *
+     * @var array
+     */
+    protected $resultFacet;
+
+    /**
+     * For each database facet, the key to the database name.
+     *
+     * @var string
+     */
+    protected $resultFacetNameKey;
+
+    /**
+     * Databases listed in configuration file
+     *
+     * @var array
+     */
+    protected $configFileDatabases;
+
+    /**
+     * Configuration of whether to use LibGuides as a data source
+     *
+     * @var bool
+     */
+    protected $useLibGuides;
+
+    /**
+     * LibGuides connector
+     *
+     * @var LibGuides
+     */
+    protected $libGuides;
+
+    /**
+     * Constructor
+     *
+     * @param \VuFind\Config\PluginManager $configManager Config PluginManager
+     * @param LibGuides                    $libGuides     LibGuides API connection
+     * @param CacheAdapter                 $cache         Object cache
+     */
+    public function __construct(
+        \VuFind\Config\PluginManager $configManager,
+        LibGuides $libGuides,
+        CacheAdapter $cache
+    ) {
+        $this->configManager = $configManager;
+        $this->libGuides = $libGuides;
+        $this->setCacheStorage($cache);
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        $settings = explode(':', $settings);
+        $this->limit
+            = (isset($settings[0]) && is_numeric($settings[0]) && $settings[0] > 0)
+            ? intval($settings[0]) : 5;
+        $databasesConfigFile = $settings[1] ?? 'EDS';
+
+        $databasesConfig = $this->configManager->get($databasesConfigFile);
```

Comment:
I wonder if it's worth creating a shortcut variable to $databasesConfig->Databases so you can shorten all your various configuration references below. Not required, just a thought for readability.

Reply:
Yes agreed, I simplified those references and threw an earlier exception if the section is not found.